02-library/cs506/sim.py

```diff
@@ -1,13 +1,27 @@
+
+def sqrt(x):
+    return x**(1/2)
+
 def euclidean_dist(x, y):
-    raise NotImplementedError()
+    return sqrt(sum([(i[0] - i[1])**2 for i in zip(x,y)]))
 
 def manhattan_dist(x, y):
-    raise NotImplementedError()
+    return sum([abs(i[0] - i[1]) for i in zip(x,y)])
 
 def jaccard_dist(x, y):
-    raise NotImplementedError()
+
+    union = len(list(set.union(set(x), set(y))))
+    inter = len(list(set.intersection(set(x), set(y))))
+    
+    if union == 0 or inter == 0:
```

Comment:
```suggestion
    if union == 0 :
```
does this make sense?